### PR TITLE
Uncommitted GC write deleted additionally as text

### LINKED ISSUE #4677: Uncommitted GC backup & restore

### PR DESCRIPTION
**Pull request by nopcoder**
Close #4677

### CODE REVIEW

**Review by N-o-Z — APPROVED** [review]
LGTM!

File: clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala [hunk 1]
```diff
@@ -221,7 +221,9 @@ object UncommittedGarbageCollector {
     val reportDst = reportPath(storageNamespace, runID)
     writeJsonSummary(reportDst, runID, firstSlice, startTime, success, expiredAddresses.count())
 
-    expiredAddresses.write.parquet(s"${reportDst}/deleted")
+    val cachedAddresses = expiredAddresses.cache()
```
Comment (N-o-Z) [inline clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala:224]: Can we add a comment or TODO here, that we need to remove the caching + text file once we have a working backup solution?

Comment (nopcoder) [reply clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala:224]: Will open an issue - thanks!

Comment (nopcoder) [reply clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala:224]: https://github.com/treeverse/lakeFS/issues/4794

**Pull request merged**